docs/torch/link_pred.rst

```diff
@@ -16,83 +16,83 @@ We generate this dataset with networkx, save it to json format and use snark to
 
 .. code-block:: python
 
-	>>> import networkx as nx
-	>>> import random
-	>>> random.seed(0)
-	>>> num_clusters = 30
-	>>> num_nodes_in_cluster = 12
-	>>> g = nx.connected_caveman_graph(num_clusters, num_nodes_in_cluster)
-	>>> test_ratio = 0.2 # Ratio of edges in a test dataset
-	>>> edge_list = list(nx.edges(g))
-	>>> random.shuffle(edge_list)
-	>>> test_cutoff = int(test_ratio * len(edge_list))
-	>>> test_dataset = set(edge_list[:test_cutoff])
-	>>> train_dataset = set(edge_list[test_cutoff:])
-	>>> import os
-	>>> working_dir = "/tmp/caveman"
-	>>> try:
-	...     os.mkdir(working_dir)
-	... except FileExistsError:
-	...     pass
-	>>> import json
-	>>> nodes = []
-	>>> data = ""
-	>>> for node_id in g:
-	...     cluster_id = float(node_id // num_nodes_in_cluster)
-	...     train_list = []
-	...     test_list = []
-	...     for neighbor_id in nx.neighbors(g, node_id):
-	...         if (node_id, neighbor_id) in train_dataset:
-	...             train_list.append(neighbor_id)
-	...         else:
-	...             test_list.append(neighbor_id)
-	...     node = {
-	...         "node_weight": 1,
-	...         "node_id": node_id,
-	...         "node_type": 0,
-	...         "uint64_feature": None,
-	...         "float_feature": {
-	...             "0": [float(cluster_id)/num_clusters],
-	...         },
-	...         "binary_feature": None,
-	...         "edge": [
-	...             {
-	...                 "src_id": node_id,
-	...                 "dst_id": neighbor_id,
-	...                 "edge_type": 0,
-	...                 "weight": 1.0,
-	...             }
-	...             for neighbor_id in train_list
-	...         ] + [
-	...             {
-	...                 "src_id": node_id,
-	...                 "dst_id": neighbor_id,
-	...                 "edge_type": 1,
-	...                 "weight": 1.0,
-	...             }
-	...             for neighbor_id in test_list
-	...         ],
-	...         "neighbor": {
-	...             "0": dict([(str(neighbor_id), 1.0) for neighbor_id in train_list]),
-	...             "1": dict([(str(neighbor_id), 1.0) for neighbor_id in test_list])
-	...         },
-	...     }
-	...     data += json.dumps(node) + "\n"
-	...     nodes.append(node)
-	>>> data_filename = working_dir + "/data.json"
-	>>> with open(data_filename, "w+") as f:
-	...     f.write(data)
-	357456
-
-	>>> import deepgnn.graph_engine.snark.convert as convert
-	>>> from deepgnn.graph_engine.snark.decoders import JsonDecoder
-	>>> partitions = 1
-	>>> convert.MultiWorkersConverter(
-	...     graph_path=data_filename,
-	...     partition_count=partitions,
-	...     output_dir=working_dir,
-	...     decoder=JsonDecoder,
-	... ).convert()
+    >>> import networkx as nx
```

Comment:
why do you need to change spacing?

Reply:
They were tabs before which was causing issues since new lines were spaces and those old ones were tabs.